docs/contributing.md

```diff
@@ -40,29 +40,27 @@ You'll need to have a version between **Python 3.7 and 3.11**, **virtualenv**, *
 git clone git@github.com:<your username>/pydantic.git
 cd pydantic
 
-# 2. Set up a virtualenv for running tests
-virtualenv -p `which python3.8` env
-source env/bin/activate
-# Building docs requires 3.8. If you don't need to build docs you can use
-# whichever version; 3.7 will work too.
+# 2. Install PDM and pre-commit
+# We use pipx here, for other options see:
+# https://pdm.fming.dev/latest/#installation
+# https://pre-commit.com/#install
+# To get pipx itself:
+# https://pypa.github.io/pipx/
+pipx install pdm pre-commit
```

Comment:
I think simpler to use `python -m venv` and `pip` here since that's the tool most people will understand and have installed.

Reply:
you could add a comment saying "pipx also works, if that's what you're used to".

Reply:
The point is you want to install these tools _outside_ of the venv. We could say `pip install pdm pre-commit` but IMO using `pipx` is much nicer than global installs because of the isolation.